pkg/common/utils/ioutil_test.go

```diff
@@ -32,12 +33,20 @@ func TestIoutilCopyBuffer(t *testing.T) {
 	srcLen := int64(src.Len())
 	written, err := CopyBuffer(dst, src, buf)
 
-	if written != srcLen {
-		t.Fatalf("Unexpected written: %d. Expecting: %d", written, srcLen)
-	}
-	if err != nil {
-		t.Fatalf("Unexpected error: %s", err)
-	}
+	assert.DeepEqual(t, written, srcLen)
+	assert.DeepEqual(t, err, nil)
+}
+
+func TestIoutilCopyBufferWithNilBuffer(t *testing.T) {
+	var writeBuffer bytes.Buffer
+	src := bytes.NewBufferString("hertz is very good!!!")
+	dst := network.NewWriter(&writeBuffer)
+	// src.Len() will change, when use src.read(p []byte)
+	srcLen := int64(src.Len())
+	written, err := CopyBuffer(dst, src, nil)
+
+	assert.DeepEqual(t, written, srcLen)
+	assert.DeepEqual(t, err, nil)
```

Comment:
maybe add a check for the data correctness?

Reply:
done